### PR TITLE
Use a base class ManagerModule for all modules/manager/*.py module cl…

### PR DESCRIPTION
**Pull request by watologo1**
…asses

By this cleanup, also split up cobbler sync into dhcp and dns only sync
capabilities.

Now there are:
```
cobbler sync -> same as before
```
and:
```
cobbler sync --dhcp
cobbler sync --dns
cobbler sync --dhcp --dns
```
to only sync dhcp and dns or only one of these.

This allows a huge speed-up if syncing is only needed, if a host got
added or deleted or a network interface and cobbler shall only re-create
dns and/or dhcp configs.

`get_manager()` now returns a single (Singleton) instance of a module, instead of creating new objects every time.
`module_loader.py:get_module_from_file()` is called by `cobbler/api.py`

### CODE REVIEW

**Comment by watologo1** [comment]
Puhh..., this Codacy code checker is tough.
I handled quite some (sane) checks. For the rest I am not that sure, it's possible or makes sense.

File: cobbler/modules/managers/dnsmasq.py [hunk 1]
```diff
@@ -240,4 +200,9 @@ def get_manager(collection_mgr):
     :param collection_mgr: The collection manager to resolve all information with.
     :return: The object generated from the class.
     """
-    return DnsmasqManager(collection_mgr)
+    # Singleton used, therefore ignoring 'global'
+    global MANAGER # pylint: disable=global-statement
+
+    if not MANAGER:
+        manager = _DnsmasqManager(collection_mgr)
```
Comment (nodeg) [inline cobbler/modules/managers/dnsmasq.py:207]:
```suggestion
        MANAGER = _DnsmasqManager(collection_mgr)
```

**Review by nodeg — APPROVED** [review]
LGTM. What`s your opinion, @SchoolGuy?

**Comment by SchoolGuy** [comment]
We did not write tests for the real functionality of what Thomas implemented. So we have still some - work to do @nodeg . But apart from that it is lgtm in my eyes yes.

**Comment by nodeg** [comment]
That's correct but testing and the state of our tests is a problem in general. It depends on how quick this PR and #2601 should be integrated into Cobbler.
We have a very basic CLI test for `cobbler sync`

```python
    def test_cobbler_sync(self, run_cmd, get_last_line):
        """Runs 'cobbler sync'"""
        (outputstd, outputerr) = run_cmd(cmd=["sync"])
        lines = outputstd.split("\n")
        assert "*** TASK COMPLETE ***" == get_last_line(lines)

```
which could be extended for `cobbler sync --dns`, `cobbler sync --dhcp` and `cobbler sync --dhcp --dns` like

```python
    def test_cobbler_sync_dns(self, run_cmd, get_last_line):
        """Runs 'cobbler sync --dns'"""
        (outputstd, outputerr) = run_cmd(cmd=["sync", "--dns"])
        lines = outputstd.split("\n")
        assert "*** TASK COMPLETE ***" == get_last_line(lines)

    def test_cobbler_sync_dhcp(self, run_cmd, get_last_line):
        """Runs 'cobbler sync --dhcp'"""
        (outputstd, outputerr) = run_cmd(cmd=["sync", "--dhcp"])
        lines = outputstd.split("\n")
        assert "*** TASK COMPLETE ***" == get_last_line(lines)

    def test_cobbler_sync_dhcp_dns(self, run_cmd, get_last_line):
        """Runs 'cobbler sync --dhcp --dns'"""
        (outputstd, outputerr) = run_cmd(cmd=["sync", "--dhcp", "--dns"])
        lines = outputstd.split("\n")
        assert "*** TASK COMPLETE ***" == get_last_line(lines)
```

**Review by SchoolGuy — APPROVED** [review]
LGTM

**Pull request merged**